ethcore/src/block.rs

```diff
@@ -543,14 +533,29 @@ fn enact(
 		db,
 		parent,
 		last_hashes,
-		Address::new(),
+		// Engine such as Clique will calculate author from extra_data.
+		// this is only important for executing contracts as the 'executive_author'.
+		engine.executive_author(&header),
 		(3141562.into(), 31415620.into()),
-		vec![],
+		header.extra_data().clone(),
 		is_epoch_begin,
 		ancestry,
 	)?;
 
-	b.populate_from(&header);
+	b.block.header.set_difficulty(*header.difficulty());
+	b.block.header.set_gas_limit(*header.gas_limit());
+	b.block.header.set_timestamp(header.timestamp());
+	b.block.header.set_uncles_hash(*header.uncles_hash());
+	b.block.header.set_transactions_root(*header.transactions_root());
+
+	if log_enabled!(target: "enact", ::log::Level::Trace) && s.is_some() {
+		let env = b.env_info();
+		let root = s.as_ref().unwrap().root();
+		let author_balance = s.as_ref().unwrap().balance(&env.author)?;
+		trace!(target: "enact", "num={}, root={}, author={}, author_balance={}\n",
+		       b.block.header.number(), root, env.author, author_balance);
+	}
+
```

Comment:
This much of `unwrapping` sticks in my eyes. 

If think it would be sufficient with:

```rustc
	if let Some(ref s) = trace_state {
		let env = b.env_info();
		let author_balance = s.balance(&env.author)?;
		trace!(target: "enact", "num={}, root={}, author={}, author_balance={}\n",
		       b.block.header.number(), s.root(), env.author, author_balance);
	}

Because `s` will only `Some` if the log is enabled
```